Mathlib/Topology/UnitInterval.lean

```diff
@@ -209,6 +209,20 @@ theorem mul_pos_mem_iff {a t : ℝ} (ha : 0 < a) : a * t ∈ I ↔ t ∈ Set.Icc
 theorem two_mul_sub_one_mem_iff {t : ℝ} : 2 * t - 1 ∈ I ↔ t ∈ Set.Icc (1 / 2 : ℝ) 1 := by
   constructor <;> rintro ⟨h₁, h₂⟩ <;> constructor <;> linarith
 
+/-- The unit interval as a submoniod of ℝ. -/
+def unitIntervalSubmonoid : Submonoid ℝ where
```

Comment:
Does that make the full name `unitInterval.unitIntervalSubmonoid`? That seems duplicative

Reply:
Does it work to just change the original definition, instead of making a new one? In other words, move this change to the top of the file, replacing lines 28-29.

Reply:
I tried to change the definition of `unitInterval` as a `Submonoid ℝ`.
Below there are some instance definitions and there I get
"failed to synthesize
  CompactSpace ↥I"
What should I do?

Reply:
@jcommelin `unitInterval` is a reducible definition, so we reuse all instances about `Set.Icc (0 : Real) 1`.

Reply:
LGTM, but let's wait for @Ruben-VandeVelde and @jcommelin .

Reply:
I removed the `awaiting-author` tag as the requested change seems unnecessary

Reply:
I think the name should be changed to `unitInterval.submonoid`.

As for the change in the definition, I suggest that we change `unitInterval` into a coerced subtype (i.e., `UnitInterval : Type := { carrier := Set.Icc (0 : ℝ) 1, ... : Submonoid ℝ }`, but as Yury suggests, this can be done in a separate PR.

Reply:
name changed.